src/libraries/Native/Unix/System.Native/CMakeLists.txt

```
@@ -47,6 +47,11 @@ else ()
         pal_log.c)
 endif ()
 
+if (CLR_CMAKE_TARGET_MACCATALYST OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS)
```

this should only apply to catalyst:

```suggestion
if (CLR_CMAKE_TARGET_MACCATALYST)
```